internal/jimm/model.go

```diff
@@ -742,16 +743,101 @@ func (j *JIMM) ModelInfo(ctx context.Context, user *openfga.User, mt names.Model
 	return j.mergeModelInfo(ctx, user, mi, m)
 }
 
+// modelSummariesMap is a safe map to add records concurrently because the access is guarded by a Mutex.
+// The read operations are not guarded because only inserts are done concurrently.
+type modelSummariesMap struct {
+	mu             sync.Mutex
```

Comment:
Can just use sync.Map, no need for mutex @SimoneDutto

Reply:
Ah I see, so if routine A handles keys 1-10, and routine B handled keys 11-20, then sync.Map is preferred. Never noticed the second part really.